### DISCUSSION TITLE
Future of IPS Peek (cross-platform?)

### DISCUSSION THREAD

**Discussion by vector-man** [vector-man/IPS-Peek#11] [Ideas]
UPDATE (2/4/2026): View here for future plans for IPS Peek. Moving to multiple platforms is the primary next step : https://github.com/vector-man/IPS-Peek/issues/40

ORIGINAL MESSAGE:
Just an update: I have an experimental version of IPS Peek running on a new UI framework, Avalonia. This will work with every platform (Windows, Linux, and Mac.)

Here's a screenshot (a very early UI design.)

![IPS_PEEK_15_2022.PNG](https://user-images.githubusercontent.com/4753368/185103387-1036c760-0a00-4a33-af63-1fcf7c30e0fd.PNG)

It currently lacks a hex view. I will have to wait for a usable control for the framework before I can implement that.

More to come! 

Original message:

Hey there, everyone!

I'm exploring how to get IPS Peek running on other platforms.

First idea: rewrite the application using Maui (that'll get Mac and Android taken care of): https://docs.microsoft.com/en-us/dotnet/maui/what-is-maui

Second idea: how about getting it running in the browser!

Going to see what I can do with the project soon.

Take care!

**Comment by limajdaas** [comment]
Sounds excellent! IPS Peek is incredibly useful and I'm glad that it's still getting updates! Is there anyway you could host the documentation here? The old website has been down for a while.

Have you considered adding in an IPS disassembler? It'd be pretty handy to have in the software so that you can not only see the offset of the differences and their hex but also the instructions that are there. It could possibly help with conflict resolution.

**Comment by vector-man** [comment]
Glad you enjoy IPS Peek! The disasember sounds like an awesome idea; I'll make a note of that. About the documentation: I'll see if I can get that added. Going to dig through my folders to see if I can find it. Yeah, my site has been offline for quite awhile. I had to take a break from coding for various reasons. But now, it looks like I can continue working on my projects.

**Comment by vector-man** [comment]
Looks like this may be a plan: 

1. Stick to Windows Forms and rewrite project for ReactiveUI 
2. Move to Avalonia, while keeping the current hex editor in a control host (this will still tie the project to Windows.)
3. Create a Avalonia port of WpfHexEditorControl (this will be far down, unfortunately.)

Edit:

Here's an alternative plan:

1. Stick to Windows Forms and rewrite project for ReactiveUI and try to make sure Mono works (this will let Linux be supported in some way.)
2. Create Avalonia "lite" version for mobile and use Droppers/HexEditorControl: https://github.com/Droppers/HexEditorControl
3. Eventually, if/when HexEditorControl matures enough, IPS Peek "lite" becomes full, and WinForms version becomes deprecated.

**Comment by vector-man** [comment]
View here for future plans for IPS Peek. Moving to multiple platforms is the primary next step : https://github.com/vector-man/IPS-Peek/issues/40
